server/src/server.ts

```diff
@@ -1,15 +1,21 @@
-import './utils/environment'
+import './utils/environment';
 import http from 'http';
 import { Server } from 'socket.io';
 import express from 'express';
 import { registerSocketEvents } from '@app/api/ws';
 import router from '@app/api/rest';
+import cors from 'cors';
 
+const corsOptions = {
+  origin: 'http://localhost:3001',
+};
```

Comment:
No need to specify origin (and directly potential env var value) - proposing to go with simply `app.use(cors());`